src/its_preselector/preselector.py

```diff
@@ -77,62 +79,114 @@ def __set_cal_sources(self, cal_sources):
+                    path_name=rf_path_name))
+
+    def get_frequency_high_stopband(self, rf_path_name: str) -> float:
+        """
+        Get the high frequency of the 60 dB stopband.
+        :param rf_path_name:
+        :return: The high frequency of the 60 dB stopband in Hz.
+        """
+        if rf_path_name in self.rf_paths:
+            path = self.rf_paths[rf_path_name]
             filter_id = path.filter_id
             preselctor_filter = self.__get_filter(filter_id)
             if preselctor_filter:
                 return preselctor_filter.frequency_high_stopband
-        return None
-
-    def get_amplifier_gain(self, rf_path_index):
-        if rf_path_index < len(self.rf_paths):
-            path = self.rf_paths[rf_path_index]
+        else:
+            raise ConfigurationException(
+                "Unable to get frequency_high for the stopband filter. There is no RF_PATH named {path_name}".format(
+                    path_name=rf_path_name))
+
+    def get_amplifier_gain(self, rf_path_name: str) -> float:
+        """
+        Get the gain of the amplifier in the specified rf_path.
+        :param rf_path_name:
+        :return:
+        """
+        if rf_path_name in self.rf_paths:
+            path = self.rf_paths[rf_path_name]
             amp_id = path.amplifier_id
             amplifier = self.__get_amplifier(amp_id)
             if amplifier:
                 return amplifier.gain
-        return None
-
-    def get_amplifier_noise_figure(self, rf_path_index):
-        if rf_path_index < len(self.rf_paths):
-            path = self.rf_paths[rf_path_index]
+        else:
+            raise ConfigurationException(
+                "Unable to get amplifier gain. There is no RF_PATH named {path_name}".format(
+                    path_name=rf_path_name))
+
+    def get_amplifier_noise_figure(self, rf_path_name: str) -> float:
+        """
+        Get the noise figure of the amplifier in the specified rf_path.
+        :param rf_path_name: The name of the rf_path.
+        :return: the noise figure of the amplifier in the specified rf_path.
+        """
+        if rf_path_name in self.rf_paths:
+            path = self.rf_paths[rf_path_name]
             amp_id = path.amplifier_id
             amplifier = self.__get_amplifier(amp_id)
             if amplifier:
                 return amplifier.noise_figure
-        return None
+        else:
+            raise ConfigurationException(
+                "Unable to get amplifier noise figure. There is no RF_PATH named {path_name}".format(
+                    path_name=rf_path_name))
 
     @abstractmethod
-    def set_state(self, i):
+    def set_state(self, state_name: str) -> None:
+        """
+        Sets the state of the preselector.
+        :param state_name: The name of the state (config key value) to enable in the preselector.
+        """
         pass
 
     def __get_filter(self, filter_id):
```

Comment:
Why not use another `@abstractmethod`? Is this just to allow preselectors without filters?

Reply:
If you mean for __get_filter this way provides an implementation for any inheriting class.

Reply:
Yeah, this was in reference to __get_filter. Is it better to have an implementation which always returns None instead of making it an abstractmethod? I guess I don't see the use case for inheriting this instead of implementing it within an inheriting class.